code/game/objects/structures/window.dm

```diff
@@ -363,11 +363,9 @@
 	move_update_air(T)
 
 /obj/structure/window/CanAtmosPass(turf/T)
-	if(get_dir(loc, T) == dir)
-		return !density
-	if(dir == FULLTILE_WINDOW_DIR)
-		return !density
-	return 1
+	if(!anchored || !density)
+		return TRUE
+	return !(get_dir(loc, T) == dir || dir = FULLTILE_WINDOW_DIR)
```

Comment:
since you made a mistake here, let's swap the order you have them here so that it shortcircuits on the most common case
`return !(dir == FULLTILE_WINDOW_DIR || get_dir(loc, T) == dir)`